pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -242,25 +244,65 @@ func (c *Controller) updateServiceNodePortAddresses(svcs ...*model.Service) bool
 		return false
 	}
 	for _, svc := range svcs {
+		if c.isNodePortGatewayService(svc) && svc.Attributes.ExternalTrafficPolicy == model.ExternalTrafficPolicyLocal {
+			endpoints := c.endpoints.buildIstioEndpointsWithService(svc.Attributes.Name, svc.Attributes.Namespace, svc.Hostname)
+			if features.EnableK8SServiceSelectWorkloadEntries {
+				fep := c.collectWorkloadInstanceEndpoints(svc)
+				endpoints = append(endpoints, fep...)
+			}
+			nodesWithEndpoints := getWorkloadNodeLocations(endpoints)
+			log.Debugf("---> svc:%s nodesWithEndpoints = %#v", svc.Hostname, nodesWithEndpoints)
```

Comment:
nit: remove or change it to correct format

Reply:
Removed it.